### PR TITLE
Use Domenic's ESLint config as a base

### PR DESCRIPTION
**Pull request by domenic**

### CODE REVIEW

**Review by TimothyGu — APPROVED** [review]
Looks good. Do we want to start using Prettier or some other code formatter?

File: lib/types.js [hunk 1]
```diff
@@ -83,7 +94,14 @@ function deepClone(obj) {
...
 }
 
-function generateTypeConversion(ctx, name, idlType, argAttrs = [], parentName, errPrefix = '"The provided value"') {
+function generateTypeConversion(
+  ctx,
+  name,
+  idlType,
+  argAttrs = [],
+  parentName = undefined,
+  errPrefix = '"The provided value"'
```
Comment (TimothyGu) [inline lib/types.js:103]: I'm personally a fan of trailing commas too

**Comment by domenic** [comment]
I'd be willing to use Prettier if it didn't diverge too much from my style (or could be configured as such). I haven't investigated...

File: lib/types.js (hunk 1 above)
Comment (domenic) [reply lib/types.js:103]: I can't bring myself to do it yet...

**Pull request merged**